### DISCUSSION TITLE
Not able to use app.dao().db() in hooks

### DISCUSSION THREAD

**Discussion by stevengpn** [pocketbase/pocketbase#3493] [Q&A]
I created main.pb.js in the pb_hooks folder, and gets excuted with error: Failed to execute main.pb.js:
 - TypeError: Cannot read property 'db' of undefined or null at <eval>:12:9(15)

`$app.dao().db().newQuery("SELECT id FROM placements LIMIT 100").all(result) `

The binary used for go service comes from :  pocketbase_0.18.10_windows_amd64.zip

Thanks in advance

**Comment by ganigeorgiev** [comment]
Are you running the above in a hook?

The above error indicates that the database connection is not initialized yet (the app instance is not initialized before app bootstrap). You can try:

```js
// pb_hooks/main.pb.js

onAfterBootstrap((e) => {
    // do something...
})
```
